v4/export/sql.go

```diff
@@ -642,7 +651,14 @@ func createConnWithConsistency(ctx context.Context, db *sql.DB) (*sql.Conn, erro
 	query = "START TRANSACTION /*!40108 WITH CONSISTENT SNAPSHOT */"
 	_, err = conn.ExecContext(ctx, query)
 	if err != nil {
```

Comment:
Should we check the error type here?

Reply:
We could try to detect the error type and check for the error number here. However I think retrying without the optional part should give the same result. I don't think the extra complexity is needed.

```
5.5.58 127.0.0.1:3306  test  SQL  START TRANSACTION /*!40108 WITH CONSISTENT SNAPSHOT */;
ERROR: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'WITH CONSISTENT SNAPSHOT */' at line 1
5.5.58 127.0.0.1:3306  test  SQL  SELECT @@version, @@memsql_version;
+-----------+------------------+
| @@version | @@memsql_version |
+-----------+------------------+
| 5.5.58    | 7.3.13           |
+-----------+------------------+
1 row in set (0.0357 sec)
```

Reply:
I see